### PR TITLE
Add data source id field in report detail table 

### LINKED ISSUE #1413: Add data source id field in report detail table
Using in frontend just add it when report happen

### PR DESCRIPTION
**Pull request by Benzbeeb**
Fixed #1413

### CODE REVIEW

File: scan/src/subscriptions/ReportSub.re [hunk 1]
```diff
@@ -10,6 +10,7 @@ module ValidatorReport = {
   };
 
   type report_details_t = {
+    dataSourceID: int,
```
Comment (sorawit) [inline scan/src/subscriptions/ReportSub.re]: Shouldn't this be `ID.DataSource.t`? @evilpeach

Comment (evilpeach) [reply scan/src/subscriptions/ReportSub.re]: Yes

File: scan/src/components/validator/ReportsTable.re [hunk 2]
```diff
@@ -153,18 +153,20 @@ let make = (~address) =>
...
-                               </>
-                             })
-                           ->React.array}
-                        </Col>
+                        {reportDetails
+                         ->Belt_Array.map(reportDetail => {
+                             <>
+                               <Row>
+                                 <TypeID.DataSource
+                                   id={ID.DataSource.ID(reportDetail.dataSourceID)}
```
Comment (evilpeach) [inline scan/src/components/validator/ReportsTable.re]: It should be `id=reportDetail.dataSourceID`.

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by taobun — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]
LGTM

**Pull request merged**